web/packages/teleport/src/LocksV2/Locks/Locks.tsx

```diff
@@ -0,0 +1,210 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, { useState, useEffect } from 'react';
+import { useLocation, useHistory } from 'react-router';
+import { formatRelative } from 'date-fns';
+import { Danger } from 'design/Alert';
+
+import Table, { Cell } from 'design/DataTable';
+import { ButtonPrimary, Label as Pill } from 'design';
+import useAttempt from 'shared/hooks/useAttemptNext';
+
+import cfg from 'teleport/config';
+import {
+  FeatureBox,
+  FeatureHeader,
+  FeatureHeaderTitle,
+} from 'teleport/components/Layout';
+import { NavLink } from 'teleport/components/Router';
+import useTeleport from 'teleport/useTeleport';
+
+import { lockService, Lock, LockTarget } from 'teleport/services/locks';
+
+import { TrashButton } from '../common';
+
+import { DeleteLockDialogue } from './DeleteLockDialogue';
+
+export function Locks() {
+  const ctx = useTeleport();
+  const history = useHistory();
+  const location = useLocation<{ createdLocks: Lock[] }>();
+  const { attempt, run } = useAttempt();
+  const [locks, setLocks] = useState<Lock[]>([]);
+  const [lockToDelete, setLockToDelete] = useState<Lock>();
+
+  useEffect(() => {
+    run(() =>
+      lockService.fetchLocks().then(res => {
+        const updatedLocks = [...res];
+        // If location state was set, user is coming back from
+        // creating a set of locks. Because of possible cache lagging,
+        // we will manually add missing new locks to the fetched list.
+        if (location.state?.createdLocks) {
+          const seenLock = {};
+          updatedLocks.forEach(lock => (seenLock[lock.name] = true));
+          location.state.createdLocks.forEach(lock => {
+            if (!seenLock[lock.name]) {
+              updatedLocks.push(lock);
+            }
+          });
+          history.replace({ state: {} }); // Clear loc state afterwards.
+        }
+        setLocks(updatedLocks);
+      })
+    );
+  }, []);
+
+  function deleteLock(lockName: string) {
+    return lockService.deleteLock(lockName).then(() => {
+      // Manually remove from the initial fetched
+      // list since there could be cache lagging.
+      const updatedLocks = locks.filter(lock => lock.name !== lockName);
+      setLocks(updatedLocks);
+      setLockToDelete(null);
+    });
+  }
+
+  const lockAccess = ctx.storeUser.getLockAccess();
+  const canCreate = lockAccess.create && lockAccess.edit;
+
+  return (
+    <>
+      <FeatureBox>
+        <FeatureHeader>
+          <FeatureHeaderTitle>Session & Identity Locks</FeatureHeaderTitle>
+          <ButtonPrimary
+            as={NavLink}
+            to={cfg.getNewLocksRoute()}
+            ml="auto"
+            disabled={!canCreate}
+            title={
+              canCreate
+                ? ''
+                : 'You do not have access to create and update locks'
```

Comment:
Can we update locks?

Reply:
you can https://github.com/gravitational/teleport/blob/860d91012c2a2087a58861a2e20682300b2eb7ea/lib/auth/auth_with_roles.go#L4876 (but no exposed api for the web to use atm), it's an `upsert` action that requires user to have both the create and update rule